src/bid/score.rs

```diff
@@ -5,22 +5,107 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 //! Score generation
-#![cfg(feature = "std")]
 
-use super::Score;
+#[cfg(feature = "canon")]
+use canonical::Canon;
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
+
+#[cfg(feature = "std")]
 use crate::bid::Bid;
+#[cfg(feature = "std")]
 use crate::errors::BlindBidError;
+use core::ops::Deref;
 use dusk_bls12_381::BlsScalar;
-use dusk_bytes::Serializable;
+use dusk_bytes::{DeserializableSlice, Serializable};
+#[cfg(feature = "std")]
 use dusk_jubjub::JubJubAffine;
+#[cfg(feature = "std")]
 use dusk_plonk::prelude::*;
+#[cfg(feature = "std")]
 use num_bigint::BigUint;
+#[cfg(feature = "std")]
 use num_traits::{One, Zero};
+#[cfg(feature = "std")]
 use plonk_gadgets::{
     AllocatedScalar, RangeGadgets::max_bound, ScalarGadgets::maybe_equal,
 };
+#[cfg(feature = "std")]
 use poseidon252::sponge;
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Default)]
+#[cfg_attr(feature = "canon", derive(Canon))]
+/// The `Score` represents a "random" value obtained from the computations
+/// based on blockchain data as well as [Bid](self::Bid) data.
+/// It derefs to it's value although the structure contains more fields which
+/// are side-results of this computation needed to proof the correctness of the
+/// Score generation process later on.
+pub struct Score {
+    value: BlsScalar,
+    y: BlsScalar,
+    y_prime: BlsScalar,
+    r1: BlsScalar,
+    r2: BlsScalar,
+}
+
+impl Deref for Score {
+    type Target = BlsScalar;
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
+
+impl AsRef<BlsScalar> for Score {
+    fn as_ref(&self) -> &BlsScalar {
+        &self.value
+    }
+}
+
+impl Serializable<{ 5 * BlsScalar::SIZE }> for Score {
+    type Error = dusk_bytes::Error;
+
+    fn to_bytes(&self) -> [u8; Self::SIZE] {
+        let mut buf = [0u8; Self::SIZE];
+        buf[0..BlsScalar::SIZE].copy_from_slice(&self.as_ref().to_bytes());
+        buf[BlsScalar::SIZE..BlsScalar::SIZE * 2]
+            .copy_from_slice(&self.as_ref().to_bytes());
+        buf[BlsScalar::SIZE * 2..BlsScalar::SIZE * 3]
+            .copy_from_slice(&self.as_ref().to_bytes());
+        buf[BlsScalar::SIZE * 3..BlsScalar::SIZE * 4]
+            .copy_from_slice(&self.as_ref().to_bytes());
+        buf[BlsScalar::SIZE * 4..Self::SIZE]
+            .copy_from_slice(&self.as_ref().to_bytes());
+        buf
+    }
+
+    fn from_bytes(buf: &[u8; Self::SIZE]) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Score {
+            value: BlsScalar::from_slice(&buf[0..BlsScalar::SIZE])?,
```

Comment:
It's fine to merge as is, but maybe you want to wait for the `from_reader` method?

Reply:
No, following our guidelines we should merge and do another PR later on.

Reply:
Will add an issue for that

Reply:
Added #105 for that

Reply:
Which guidelines are you referring to? Here we're talking about two parallel changes that are happened at the same time – actually the PR for `from_reader` was done yesterday – so I'm not sure at which part of the guidelines you're referring to.

I'm fine to have it implemented in a follow up issue, as mentioned to me the only mandatory change on the first PR was the naming convention; but I want to be sure we're on the same page about the guidelines.